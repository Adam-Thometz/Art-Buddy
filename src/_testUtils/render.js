import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
