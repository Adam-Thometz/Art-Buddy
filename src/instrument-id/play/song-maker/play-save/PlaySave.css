.PlaySave {
  width: 33%;
}

.PlaySave button {
  width: 50%;
}
