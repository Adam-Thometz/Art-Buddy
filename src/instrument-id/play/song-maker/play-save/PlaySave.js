import React from 'react';

import { useSelector, useDispatch } from 'react-redux';
import { toggleLoop } from '_redux/instrument-id/song-maker/songMakerActions';

import './PlaySave.css';

import Button from '_components/button/Button';
import Popup from '_components/popup/Popup';
import SaveSong from '../save-song/SaveSong';

const PlaySave = () => {
  const { song, isPlaying } = useSelector(state => state.songMaker);
  const dispatch = useDispatch();

  const handleToggle = () => {
    if (song.some(part => part || !!part.melodyId)) dispatch(toggleLoop());
  };

  return (
    <div className='PlaySave'>
      <Button colorId={isPlaying ? 2 : 0} onClick={handleToggle}>
        {isPlaying ? 'STOP' : 'PLAY'}
      </Button>
      <Popup
        title='SAVE SONG'
        trigger={<Button colorId={3}>SAVE</Button>}
        popup={<SaveSong />}
      />
    </div>
  )
}

export default PlaySave