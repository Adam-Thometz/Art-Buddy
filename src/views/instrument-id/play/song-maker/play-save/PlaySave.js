import { useContext, useEffect } from "react";
import { PopupContext } from "context/PopupContext";

import { useSelector, useDispatch } from "react-redux";
import { toggleLoop } from "store/instrument-id/song-maker/songMakerReducer";

import "./PlaySave.css";

import Button from "components/button/Button";
import SaveSong from "../save-song/SaveSong";

const PlaySave = () => {
  const { song, isPlaying } = useSelector((state) => state.songMaker);
  const { setCurrPopup } = useContext(PopupContext);
  const dispatch = useDispatch();

  useEffect(() => {
    return () => setCurrPopup(null);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleToggle = () => {
    if (song.some((part) => part || !!part.melodyId)) dispatch(toggleLoop());
  };

  const openSavePopup = () =>
    setCurrPopup({
      title: "SAVE SONG",
      popup: <SaveSong />,
    });

  return (
    <div className="PlaySave">
      <Button colorId={isPlaying ? 2 : 0} onClick={handleToggle}>
        {isPlaying ? "STOP" : "PLAY"}
      </Button>
      <Button colorId={3} onClick={openSavePopup}>
        SAVE
      </Button>
    </div>
  );
};

export default PlaySave;
