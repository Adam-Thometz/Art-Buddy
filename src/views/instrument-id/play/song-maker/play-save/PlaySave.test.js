import PlaySave from "./PlaySave";

import { render } from "testUtils/render";

describe("PlaySave component", () => {
  it("renders without crashing", () => {
    render(<PlaySave />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<PlaySave />);
    expect(asFragment()).toMatchSnapshot();
  });
});
