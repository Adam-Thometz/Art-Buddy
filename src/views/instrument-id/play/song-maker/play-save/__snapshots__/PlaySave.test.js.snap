// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaySave component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="PlaySave"
  >
    <button
      class="Button regular"
      style="border-color: #89eb0a;"
    >
      PLAY
    </button>
    <button
      class="Button regular"
      style="border-color: #4d4f5e;"
    >
      SAVE
    </button>
  </div>
</DocumentFragment>
`;
